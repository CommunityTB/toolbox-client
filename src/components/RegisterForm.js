import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { AppContext } from '../components/AppProvider';


class RegisterForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      first_name: `First Name`,
      last_name: `Last Name`,
      user_password: `********`,
      email: `your-email@example.com`,
      user_name: `your-username`
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  static contextType = AppContext

  handleSubmit(e) {
    e.preventDefault()
    const newUser = {
      first_name: e.target['firstName'].value,
      last_name: e.target['lastName'].value,
      email: e.target['email'].value,
      user_name: e.target['userName'].value,
      user_password: e.target['password'].value,
    }

    this.context.actions.registerUser(newUser)
    this.props.onCreateNewUser()
  }

  handleChange(e) {
    const { name, value } = e.target
    this.setState({ [name]: value });
  }

  render() {
    const { first_name, last_name, user_name, user_password, email } = this.state
    return (
      <form onSubmit={this.handleSubmit}>
      <div className="form-section">
        <div className="label-holder"><label htmlFor="first-name">First Name</label></div>
        <input className="form-input-field" id="first-name" type="text" name="firstName" placeholder={first_name} onChange={this.handleChange} required />
      </div>

      <div className="form-section">
        <div className="label-holder"><label htmlFor="last-name">Last Name</label></div>
        <input className="form-input-field" id="last-name" type="text" name="lastName" placeholder={last_name} onChange={this.handleChange} required />
      </div>

      <div className="form-section">
        <div className="label-holder"><label htmlFor="email">Email</label></div>
        <input className="form-input-field" id="email" type="text" name="email" placeholder={email} onChange={this.handleChange} required />
      </div>

      <div className="form-section">
        <div className="label-holder"><label htmlFor="user-name">Username</label></div>
        <input className="form-input-field" id="user-name" type="text" name="userName" placeholder={user_name} onChange={this.handleChange} required />
      </div>

      <div className="form-section">
        <div><label htmlFor="password">Password</label></div>
        <input className="form-input-field" id="password" type="password" name="password" placeholder={user_password} onChange={this.handleChange} required />
      </div>

      <button type="submit">Submit</button>
      <Link className="cancel-action" to='/'>Cancel</Link>
    </form>
    )
  }
}

export default RegisterForm;