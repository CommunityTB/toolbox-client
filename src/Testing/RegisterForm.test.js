import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RegisterForm from '../components/RegisterForm'

describe(`RegisterForm component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><RegisterForm /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a RegisterForm by default', () => {
    const wrapper = shallow(<RegisterForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})