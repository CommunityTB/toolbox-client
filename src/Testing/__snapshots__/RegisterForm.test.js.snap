// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm component renders a RegisterForm by default 1`] = `
<div
  className="form-space"
>
  <form
    className="register-form"
    onSubmit={[Function]}
  >
    <legend
      className="page-title"
    >
      Register
    </legend>
    <label
      className="first-name"
      htmlFor="first-name"
    >
      First Name
    </label>
    <Input
      className="form-input-field"
      id="first-name"
      name="firstName"
      onChange={[Function]}
      required={true}
      type="text"
    />
    <label
      className="last-name"
      htmlFor="last-name"
    >
      Last Name
    </label>
    <Input
      className="form-input-field"
      id="last-name"
      name="lastName"
      onChange={[Function]}
      required={true}
      type="text"
    />
    <label
      className="email"
      htmlFor="email"
    >
      Email
    </label>
    <Input
      className="form-input-field"
      id="email"
      name="email"
      onChange={[Function]}
      required={true}
      type="text"
    />
    <label
      className="user-name"
      htmlFor="user-name"
    >
      Username
    </label>
    <Input
      className="form-input-field"
      id="user-name"
      name="userName"
      onChange={[Function]}
      required={true}
      type="text"
    />
    <label
      className="user-password"
      htmlFor="password"
    >
      Password
    </label>
    <Input
      className="form-input-field"
      id="password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
    />
    <Button
      className="register-button"
      type="submit"
    >
      Submit
    </Button>
    <Link
      className="cancel-action"
      to="/"
    >
      Cancel
    </Link>
  </form>
</div>
`;
